src/data/dataset_interface.py

```diff
@@ -0,0 +1,28 @@
+import torchvision.transforms as transforms
+
+from abc import ABC, abstractmethod
+
+class TaskDataset(ABC):
+    @abstractmethod
+    def __init__(self,root_dir, kernel_size, sigma, size, batch_size=4, num_workers=2):
+        self.root_dir = root_dir
+        self.kernel_size = kernel_size
```

Comment:
Can we rename this blur_kernel_size throughout the PR? Convolutions use this name so it can be confusing